### DISCUSSION TITLE
ImportError

### DISCUSSION THREAD

**Discussion by cellphys** [miniscope/minian#208] [Q&A]
Hi there,

I tried to run minian after installation and encountered ImportError. It occurs on my two computers. Can anyone tell me what happens and what I should do? Thanks!
----

ImportError                               Traceback (most recent call last)
Input In [3], in <cell line: 31>()
     23 from minian.preprocessing import denoise, remove_background
     24 from minian.utilities import (
     25     TaskAnnotation,
     26     get_optimal_chk,
   (...)
     29     save_minian,
     30 )
---> 31 from minian.visualization import (
     32     CNMFViewer,
     33     VArrayViewer,
     34     generate_videos,
     35     visualize_gmm_fit,
     36     visualize_motion,
     37     visualize_preprocess,
     38     visualize_seeds,
     39     visualize_spatial_update,
     40     visualize_temporal_update,
     41     write_video,
     42 )

File ~\Documents\CellPhys\miniscope\minian-master\minian\visualization.py:16, in <module>
     14 import numpy as np
     15 import pandas as pd
---> 16 import panel as pn
     17 import param
     18 import scipy.sparse as scisps

File ~\.conda\envs\minian\lib\site-packages\panel\__init__.py:5, in <module>
      1 from __future__ import absolute_import, division, unicode_literals
      3 import param as _param
----> 5 from . import layout # noqa
      6 from . import links # noqa
      7 from . import pane # noqa

File ~\.conda\envs\minian\lib\site-packages\panel\layout.py:22, in <module>
     19 from bokeh.models.widgets import Tabs as BkTabs, Panel as BkPanel
     21 from .util import param_name, param_reprs
---> 22 from .viewable import Layoutable, Reactive
     24 _row = namedtuple("row", ["children"])
     25 _col = namedtuple("col", ["children"])

File ~\.conda\envs\minian\lib\site-packages\panel\viewable.py:21, in <module>
     17 from functools import partial
     19 import param
---> 21 from bokeh.document.document import Document as _Document, _combine_document_events
     22 from bokeh.document.events import ModelChangedEvent
     23 from bokeh.io import curdoc as _curdoc

File ~\.conda\envs\minian\lib\site-packages\bokeh\document\__init__.py:43, in <module>
     33 __all__ = (
     34     'DEFAULT_TITLE',
     35     'Document',
     36     'without_document_lock',
     37 )
     39 #-----------------------------------------------------------------------------
     40 # General API
     41 #-----------------------------------------------------------------------------
---> 43 from .document import DEFAULT_TITLE ; DEFAULT_TITLE
     44 from .document import Document ; Document
     45 from .locking import without_document_lock ; without_document_lock

File ~\.conda\envs\minian\lib\site-packages\bokeh\document\document.py:49, in <module>
     47 from ..core.json_encoder import serialize_json
     48 from ..core.query import find
---> 49 from ..core.templates import FILE
     50 from ..core.validation import check_integrity
     51 from ..events import Event

File ~\.conda\envs\minian\lib\site-packages\bokeh\core\templates.py:43, in <module>
     40 import sys
     42 # External imports
---> 43 from jinja2 import Environment, Markup, FileSystemLoader
     45 # Bokeh imports
     46 
     47 #-----------------------------------------------------------------------------
     48 # Globals and constants
     49 #-----------------------------------------------------------------------------
     51 __all__ = (
     52     'AUTOLOAD_JS',
     53     'AUTOLOAD_NB_JS',
   (...)
     65     'SCRIPT_TAG',
     66 )

ImportError: cannot import name 'Markup' from 'jinja2' (C:\Users\user\.conda\envs\minian\lib\site-packages\jinja2\__init__.py)

**Comment by mberensc** [comment]
see this reply in issues https://github.com/denisecailab/minian/issues/195#issuecomment-1087869025

The issue is with the versions of Bokeh, Jinja2, and markupsafe
